backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/PredicatesHandler.java

```diff
@@ -94,6 +94,66 @@
     return InternalFuture.failedStage(new InvalidArgumentException("Predicate cannot be handled"));
   }
 
+  private InternalFuture<QueryFilterContext> processTagsPredicate(
+      long index, KeyValueQuery predicate) {
+    try {
+      final var value = predicate.getValue();
+      var operator = predicate.getOperator();
+      //  We will manage this operator at bottom of method based on `NE` and `NOT_CONTAIN` using
+      // `IN` OR `NOT IN` query
+      if (operator.equals(OperatorEnum.Operator.NOT_CONTAIN)) {
+        operator = OperatorEnum.Operator.CONTAIN;
+      } else if (operator.equals(OperatorEnum.Operator.NE)) {
+        operator = OperatorEnum.Operator.EQ;
+      }
+      final var finalOperator = operator;
+
+      final var valueBindingName = String.format("v_t_%d", index);
+
+      var sql = "select distinct experiment_run_id from tag_mapping where entity_name=:entity_name";
+      sql += " and ";
+
+      final var colValue = "tags";
+      var queryContext =
+          new QueryFilterContext().addBind(q -> q.bind("entity_name", "ExperimentRunEntity"));
```

Comment:
This will collide with other `entity_name`. We have to make the bindings we create here unique (so we have to use the index). It might be a bug in my current code too.

Reply:
created unique using index and fixed binding